src/Nethermind/Nethermind.Evm.Test/BeaconParentBlockRootPrecompileTests.cs

```diff
@@ -0,0 +1,150 @@
+using Nethermind.Db;
+using Nethermind.Evm.TransactionProcessing;
+using Nethermind.Int256;
+using Nethermind.Logging;
+using Nethermind.Specs;
+using Nethermind.Specs.Forks;
+using Nethermind.State.Repositories;
+using Nethermind.State;
+using Nethermind.Trie.Pruning;
+using NUnit.Framework;
+using Nethermind.Core.Crypto;
+using Nethermind.Evm.Tracing.GethStyle;
+
+namespace Nethermind.Evm.Test;
+
+public class Eip4788Tests
+{
+    private WorldState _testState;
+    private ISpecProvider specProvider;
+    private SenderRecipientAndMiner senderRecipientAndMiner = SenderRecipientAndMiner.Default;
+    protected static IEnumerable<(IReleaseSpec Spec, bool ShouldFail)> BeacondStateRootGetPayloadV3ForDifferentSpecTestSource()
+    {
+        yield return (Shanghai.Instance, true);
+        yield return (Cancun.Instance, false);
+    }
+
+    [TestCaseSource(nameof(BeacondStateRootGetPayloadV3ForDifferentSpecTestSource))]
+    public void ParentBeaconBlockRoot_Is_Stored_Correctly_and_Only_Valid_PostCancun((IReleaseSpec Spec, bool ShouldFail) testCase)
+    {
+        specProvider = new TestSpecProvider(testCase.Spec);
+        DbProvider dbProvider = new(DbModeHint.Mem);
+        dbProvider.RegisterDb(DbNames.BlockInfos, new MemDb());
+        dbProvider.RegisterDb(DbNames.Blocks, new MemDb());
+        dbProvider.RegisterDb(DbNames.Headers, new MemDb());
+        dbProvider.RegisterDb(DbNames.State, new MemDb());
+        dbProvider.RegisterDb(DbNames.Code, new MemDb());
+        dbProvider.RegisterDb(DbNames.Metadata, new MemDb());
+        BlockTree blockTree = new(
+            dbProvider,
+            new ChainLevelInfoRepository(dbProvider),
+            specProvider,
+            NullBloomStorage.Instance,
+            LimboLogs.Instance);
+        TrieStore trieStore = new(
+            dbProvider.RegisteredDbs[DbNames.State],
+            NoPruning.Instance,
+            Archive.Instance,
+            LimboLogs.Instance);
+        _testState = new(
+            trieStore,
+            dbProvider.RegisteredDbs[DbNames.Code],
+            LimboLogs.Instance);
+        StateReader stateReader = new(trieStore, dbProvider.GetDb<IDb>(DbNames.State), LimboLogs.Instance);
+        BlockhashProvider blockhashProvider = new(blockTree, LimboLogs.Instance);
+
+        VirtualMachine virtualMachine = new(
+            blockhashProvider,
+            specProvider,
+            LimboLogs.Instance);
+        TransactionProcessor txProcessor = new(
+            specProvider,
+            _testState,
+            virtualMachine,
+            LimboLogs.Instance);
+        BlockProcessor blockProcessor = new(
+            specProvider,
+            Always.Valid,
+            NoBlockRewards.Instance,
+            new BlockProcessor.BlockValidationTransactionsExecutor(txProcessor, _testState),
+            _testState,
+            NullReceiptStorage.Instance,
+            NullWitnessCollector.Instance,
+            LimboLogs.Instance);
```

Comment:
Kind of looks like you are setting up a blockchain, don't we have that covered already by helper classes?